src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/AbiphoneFavourites.java

```diff
@@ -0,0 +1,254 @@
+						!isAbiphoneShowOnlyFavourites();
+					lastClick = System.currentTimeMillis();
+				}
+			}
+			event.setCanceled(true);
+			return;
+		}
+
+		if (!isContact(stack)) return;
+
+		int clickedButton = event.clickedButton;
+		//prevents removing the contact
+		if (clickedButton == 1) {
+			event.setCanceled(true);
+			return;
+		}
+
+		String rawName = stack.getDisplayName();
+		String name = StringUtils.cleanColour(rawName);
+		int clickType = event.clickType;
+
+		//prevents calling non favourite contacts
+		if (clickedButton == 0 && (clickType == 0 || clickType == 6)) {
+			if (!getFavouriteContacts().contains(name)) {
+				event.setCanceled(true);
+				return;
+			}
+		}
+
+		//toggle favourite contact
+		if (clickType == 1) {
+			if (!isAbiphoneShowOnlyFavourites()) {
+				toggleFavouriteContact(rawName, name);
+				event.setCanceled(true);
+			}
+		}
+	}
+
+	private void toggleFavouriteContact(String rawName, String name) {
+		if (getFavouriteContacts().contains(name)) {
+			getFavouriteContacts().remove(name);
+			Utils.addChatMessage("§e[NEU] Removed §r" + rawName + " §efrom your favourite contacts!");
+		} else {
+			getFavouriteContacts().add(name);
+			Utils.addChatMessage("§e[NEU] Added §r" + rawName + " §eto your favourite contacts!");
+		}
+	}
+
+	public boolean onRenderStack(ItemStack stack) {
+		if (isWrongInventory()) return false;
+
+		if (stack == null || stack.getDisplayName() == null) return false;
+
+		if (!isContact(stack)) return false;
+
+		String rawName = stack.getDisplayName();
+		String name = StringUtils.cleanColour(rawName);
+
+		return isAbiphoneShowOnlyFavourites() && !getFavouriteContacts().contains(name);
+	}
+
+	public void onDrawBackground(GuiScreen screen) {
+		if (isWrongInventory()) return;
+
+		GuiContainer container = (GuiContainer) screen;
+
+		for (Slot slot : container.inventorySlots.inventorySlots) {
+			if (slot == null) continue;
+			ItemStack stack = slot.getStack();
+			if (stack == null) continue;
+
+			if (isBorder(stack)) {
+				if (slot.slotNumber > 2 && slot.slotNumber < 6) {
+					if (!isAbiphoneShowOnlyFavourites()) {
+						RenderUtils.highlightSlot(slot, Color.GREEN);
+					} else {
+						RenderUtils.highlightSlot(slot, Color.ORANGE);
+					}
+				}
+			}
+
+			if (!isContact(stack)) continue;
+
+			String rawName = stack.getDisplayName();
+			String name = StringUtils.cleanColour(rawName);
+
+			if (!isAbiphoneShowOnlyFavourites()) {
+				if (getFavouriteContacts().contains(name)) {
+					RenderUtils.highlightSlot(slot, Color.ORANGE);
+				}
+			}
+		}
+	}
+
+	private static boolean isWrongInventory() {
+		if (!NotEnoughUpdates.INSTANCE.hasSkyblockScoreboard()) return true;
+		if (!NotEnoughUpdates.INSTANCE.config.misc.abiphoneFavourites) return true;
+		if (!Utils.getOpenChestName().startsWith("Abiphone ")) return true;
+
+		return false;
+	}
+
+	private static boolean isBorder(ItemStack stack) {
+		return stack.getUnlocalizedName().startsWith("tile.thinStainedGlass.");
```

Comment:
shouldn't this be `stack.getItem() == Item.getItemFromBlock(Blocks.stained_glass_pane)` ?

Reply:
Done.